ha/core/event_analyzer/parser/parser.py

```diff
@@ -176,36 +263,51 @@ def parse_event(self, msg: str) -> HealthEvent:
+                        if current_pod_restart_val is not None and current_pod_restart_val:
+                            # If the incoming generation id doesn't match with already stored
+                            # value and assosciated pod_restart count is 1. That means,
+                            # its a pod restart case and this alert is already handled
+                            sys_health_key_val["events"][0]["specific_info"]["pod_restart"] = 0
+                            self._conf_store.update(key_to_search, json.dumps(sys_health_key_val))
+                        else:
+                            # generation id is different and pod_restart value assosciated with
+                            # this pod is also zero. Means, pod restart happened and online event
+                            # came first(before delete event). So, handle two events here itself
 
-            event = {
-                EVENT_ATTRIBUTES.EVENT_ID : event_id,
-                EVENT_ATTRIBUTES.EVENT_TYPE : event_type,
-                EVENT_ATTRIBUTES.SEVERITY : StatusMapper.EVENT_TO_SEVERITY_MAPPING[event_type],
-                EVENT_ATTRIBUTES.SITE_ID : site_id, # TODO: Should be fetched from confstore
-                EVENT_ATTRIBUTES.RACK_ID : rack_id, # TODO: Should be fetched from confstore
-                EVENT_ATTRIBUTES.CLUSTER_ID : cluster_id, # TODO: Should be fetched from confstore
-                EVENT_ATTRIBUTES.STORAGESET_ID : node_id,
-                EVENT_ATTRIBUTES.NODE_ID : node_id,
-                EVENT_ATTRIBUTES.HOST_ID : node_id,
-                EVENT_ATTRIBUTES.RESOURCE_TYPE : resource_type,
-                EVENT_ATTRIBUTES.TIMESTAMP : timestamp,
-                EVENT_ATTRIBUTES.RESOURCE_ID : node_id,
-                EVENT_ATTRIBUTES.SPECIFIC_INFO : "specific_info"
-            }
+                            # One potantial problem in sending two alerts here is if after pod
+                            # comes up on some other node, then failed alert "node" will be wrongly sent.
+                            current_alert_received = json.dumps(cluster_resource_alert)
+                            # First send the failure alert
+                            cluster_resource_alert["_event_type"] = "failed"
+                            health_event = self._create_health_event_object(cluster_resource_alert, {"generation_id": gen_id, "pod_restart": 1})
+                            health_event_list.append(health_event)
+                            # and now append the incoming online alert
+                            health_event = self._create_health_event_object(json.loads(current_alert_received), {"generation_id": cluster_resource_alert["_generation_id"],"pod_restart": 1})
+                            health_event_list.append(health_event)
+                    else:
+                        # Normal pod failure occured where node id and its assosciated machine
```

Comment:
This may or may not be a failure. If the current state == existing state, no event needs to be generated.
Existing system health logic handles this

Reply:
I have modified the code in new PR. You can check there.